src/serverenvironment.cpp

```diff
@@ -279,30 +279,67 @@ void LBMManager::applyLBMs(ServerEnvironment *env, MapBlock *block, u32 stamp)
 
 void fillRadiusBlock(v3s16 p0, s16 r, std::set<v3s16> &list)
 {
+	const s16 r2 = r * r;
 	v3s16 p;
 	for(p.X=p0.X-r; p.X<=p0.X+r; p.X++)
-		for(p.Y=p0.Y-r; p.Y<=p0.Y+r; p.Y++)
-			for(p.Z=p0.Z-r; p.Z<=p0.Z+r; p.Z++)
-			{
-				// limit to a sphere
-				if (p.getDistanceFrom(p0) <= r) {
-					// Set in list
-					list.insert(p);
-				}
-			}
+	for(p.Y=p0.Y-r; p.Y<=p0.Y+r; p.Y++)
+	for(p.Z=p0.Z-r; p.Z<=p0.Z+r; p.Z++)
+	{
+		// limit to a sphere
+		if (p.getDistanceFromSQ(p0) <= r2) {
+			// Set in list
+			list.insert(p);
+		}
+	}
+}
+
+void fillViewConeBlock(v3s16 p0,
+	const s16 r,
+	const v3f camera_pos,
+	const v3f camera_dir,
+	const float camera_fov,
+	std::set<v3s16> &list)
+{
+	v3s16 p;
+	for(p.X=p0.X-r; p.X<=p0.X+r; p.X++)
+	for(p.Y=p0.Y-r; p.Y<=p0.Y+r; p.Y++)
+	for(p.Z=p0.Z-r; p.Z<=p0.Z+r; p.Z++)
+	{
+		if (isBlockInSight(p, camera_pos, camera_dir, camera_fov, r * BS * MAP_BLOCKSIZE)) {
```

Comment:
unroll r * BS * MAP_BLOCKSIZE it's not modified and will be a little bit faster too :)

Reply:
I had a version that did this, then I assumed the compiler would deduce this anyway. (r is const)
Happy to change it.